ui/src/app/job-details/tabs/scattered-attempts/scattered-attempts.component.ts

```diff
@@ -0,0 +1,40 @@
+import {Component, Inject, ViewChild} from "@angular/core";
+import {MAT_DIALOG_DATA} from "@angular/material";
+import {JobManagerService} from "../../../core/job-manager.service";
+import {TaskShard} from "../../../shared/model/TaskShard";
+import {JobStatus} from "../../../shared/model/JobStatus";
+import {JobStatusIcon} from "../../../shared/common";
+
+@Component({
+  selector: 'jm-scattered-attempts-component',
+  templateUrl: 'scattered-attempts.component.html',
+  styleUrls: ['scattered-attempts.component.css']
+})
+export class JobScatteredAttemptsComponent {
+  @ViewChild('dialogContainer') dialogContainer: HTMLElement;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private readonly jobManagerService: JobManagerService) {
+  }
+
+  getShardAttempts(shard: TaskShard) {
+    this.jobManagerService.getShardAttempts(this.data.shardsData.taskId, this.data.shardsData.taskName, shard.shardIndex).then((response) => {
+      shard.attemptsData = response.attempts;
+    });
+  }
+
+  getStatusIcon(status: JobStatus): string {
+    return JobStatusIcon[status];
+  }
+
+  hasFailures(shard: TaskShard): boolean {
```

Comment:
This also seems duplicated with the code above for checking existence. Maybe abstract to a utils file?